azure-kusto-data/azure/kusto/data/_token_providers.py

```diff
@@ -565,7 +565,7 @@ def _get_token_impl(self) -> Optional[dict]:
         return self._valid_token_or_throw(token)
 
     def _get_token_from_cache_impl(self) -> dict:
-        token = self._msal_client.acquire_token_silent(scopes=self._scopes, account=None)
+        token = self._msal_client.acquire_token_for_client(scopes=self._scopes)

```

Comment:
You sure?
If so, let's not implement _get_token_from_cache_impl and simply return none

Reply:
Yes, see the PR I linked.
Fixed to return None.